README.md

```diff
@@ -12,3 +12,163 @@ We therefore highly recommend using [Custom Dimensions](https://matomo.org/docs/
 
 **Warning**: Depending on the database size of your Matomo this plugin may take a long time to install.
 
+# Custom Variables User Guide
+
+## What is the value of Custom Variables?
+Matomo offers a Custom Variables feature for collecting custom metrics related to your page views and visits. For example, if you want to track things such as the Tags associated with a certain page that is viewed or the user role of somebody navigating through your website. This kind of data is typically managed through your CMS (Content Management System) and is likely to be different for each site so it needs a little configuration to get started.
+
+## Custom Variables vs Custom Dimensions
+**Custom Variables serve a similar purpose to Custom Dimensions, however, [Custom Dimensions](https://matomo.org/docs/custom-dimensions/) have many [advantages over Custom Variables](https://matomo.org/faq/general/faq_21117/) so in most cases you should use Custom Dimensions.**
+
+In fact, we generally recommend that the only time that you should use Custom Variables is when you need to store more than one value for the same dimension. For example in blogging and social media systems it is common to associate multiple tags with a post. In this case you could collect all of the tags within your analytics for analysis which wouldn’t be possible with Custom Dimensions.
+
+## The Anatomy of a Custom Variable
+Custom Variables consist of four different elements, all of which are required.
+
+* **Index** - This is a unique numeric ID which is typically between 1-5 that references the data. You can hold up to 5 custom variables for each scope by default, however if you are using Matomo On-Premise, it is possible to configure it to [add more than 5 custom variables](https://matomo.org/faq/how-to/faq_17931/) for each scope.
+* **Name** - This is a user-friendly label to identify the variable. For example, **Tag**.
+* **Value** - This is the actual value of the variable. For example, if the variable is tracking tags an article about Matomo could be filed as “Marketing”, “Analytics” or both.
+* **Scope** - This defines whether the variable is page scoped i.e. attached to the page it is loaded on or visit scoped and connected with the visitor that is on the page.
+
+All of the fields are limited to a total of 200 characters.
+
+## Example Custom Variables
+To help you understand the different elements we have provided an example Variable for each of the potential scopes below.
+
+**Visit Scope Dimensions: User Role**
+
+This is useful because in many cases users can have multiple roles on a website. For example somebody who is both Staff and a Moderator.
```

Comment:
@snake14 I think you missed adding the example here
```
Visit Scope Dimensions: User Role

This is useful because in many cases users can have multiple roles on a website. For example somebody who is both Staff and a Moderator.

Sample Values:

Index: 1
Name: User Role
Value: Staff
Scope: visit


Action Scope Dimension: Tags
A common design pattern for websites on platforms like WordPress is filing content under a single “Category” while adding several “Tags” to add more specific definitions to the content. For example an article might be found in the “Marketing” category while also holding tags for “Analytics” “Matomo” and “Plugins”.

Sample Values:

Index: 1
Name: Tags
Value: Marketing
Scope: action

```